### PR TITLE
Fix/deployment/445

### PR DESCRIPTION
**Pull request by majamassarini**
Creates a new weekly validation job for "minor" gitlab instances:
- gitlab.gnome.org
- gitlab.freedesktop.org
- salsa.debian.org

Merge after: packit/deployment#484

### CODE REVIEW

File: helm-charts/packit-service-validation/templates/secret-gitlab-freedesktop.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-gitlab-freedesktop
+  namespace: {{ .Release.Namespace }}
```
Comment (jpopelka) [inline helm-charts/packit-service-validation/templates/secret-gitlab-freedesktop.yml:5]: I see the existing `templates/secret*.yml` have the `namespace` set as well, but I'm pretty sure it's not needed. OpenShift sets the namespace itself based on the namespace/project we deploy/install the Helm chart/release into. You can remove it from the existing `templates/secret*.yml` if you want.

```suggestion
```

File: helm-charts/packit-service-validation/values.yaml [hunk 2]
```diff
@@ -1,11 +1,17 @@
 deployment: ~
 
 # Cron job parameters
-cronjob:
-  # runs every day at 4 a.m.
-  schedule: "0 4 * * *"
-  # maximum duration is 4 hours
-  activeDeadlineSeconds: 14400
+cronjobs:
+  - "daily validation":
```
Comment (jpopelka) [inline helm-charts/packit-service-validation/values.yaml:5]:
```suggestion
  - id: "daily validation"
```
?

Comment (majamassarini) [reply helm-charts/packit-service-validation/values.yaml:5]: 🙈 thanks!

File: helm-charts/packit-service-validation/templates/job-run-validation.yml [hunk 3]
```diff
@@ -1,36 +1,57 @@
...
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Release.Name }}-salsadebian
+                  key: token
+{{ end }}
             - name: SENTRY_SECRET
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Release.Name }}-sentry
+                  name: {{ $.Release.Name }}-sentry
```
Comment (jpopelka) [inline helm-charts/packit-service-validation/templates/job-run-validation.yml:50]: I've learned something new :+1:

**Review by jpopelka — APPROVED** [review]
Haven't tried but LGTM

**Comment by majamassarini** [comment]
> Don't forget to [bump the version](https://github.com/packit/helm/blob/main/helm-charts/packit-service-validation/Chart.yaml#L5). (I'm wondering if we should emphasize this somewhere more, but not sure where.)

Yes sure!
I am waiting to know its tag name :)
I have not yet merged changes in deployment repo.

**Review by lbarcziova — APPROVED** [review]
lgtm :+1:

**Comment by majamassarini** [comment]
> > Don't forget to [bump the version](https://github.com/packit/helm/blob/main/helm-charts/packit-service-validation/Chart.yaml#L5). (I'm wondering if we should emphasize this somewhere more, but not sure where.)
> 
> Yes sure! I am waiting to know its tag name :) I have not yet merged changes in deployment repo.

Now I see you meant the chart version... bumping it 😅

File: helm-charts/packit-service-validation/templates/job-run-validation.yml [hunk 4]
```diff
@@ -1,36 +1,57 @@
...
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Release.Name }}-gitlabfreedesktop
+                  key: token
+            - name: SALSA_DEBIAN_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Release.Name }}-salsadebian
+                  key: token
+{{ end }}
```
Comment (jpopelka) [inline helm-charts/packit-service-validation/templates/job-run-validation.yml:46]: I've just tried to run `make template foobar` in `values/packit-service-validation/` and there are a few empty lines in the output, which I think shouldn't be a problem (as it's still a valid YAML), but can be improved:

```suggestion
{{- end }}
{{- if eq $cronjob.id "weekly-validation"}}
            - name: GITLAB_GNOME_TOKEN
              valueFrom:
                secretKeyRef:
                  name: {{ $.Release.Name }}-gitlabgnome
                  key: token
            - name: GITLAB_FREEDESKTOP_TOKEN
              valueFrom:
                secretKeyRef:
                  name: {{ $.Release.Name }}-gitlabfreedesktop
                  key: token
            - name: SALSA_DEBIAN_TOKEN
              valueFrom:
                secretKeyRef:
                  name: {{ $.Release.Name }}-salsadebian
                  key: token
{{- end }}
```

File: helm-charts/packit-service-validation/templates/job-run-validation.yml [hunk 5]
```diff
@@ -1,36 +1,57 @@
...
+                  name: {{ $.Release.Name }}-sentry
                   key: dsn
             - name: DEPLOYMENT
-              value: {{ .Values.deployment }}
+              value: {{ $.Values.deployment }}
             resources:
-              {{- toYaml .Values.resources | nindent 14 }}
+              {{- toYaml $.Values.resources | nindent 14 }}
           restartPolicy: OnFailure
+{{- end }}
```
Comment (jpopelka) [inline helm-charts/packit-service-validation/templates/job-run-validation.yml:57]: Now I've spotted that there are no [--- (three dashes)](https://yaml.org/spec/1.2.2/#22-structures) between the two cronjobs when they are generated with `make template/dryrun`.
I think that might cause the k8s/Openshift to merge it somehow into one job.

```suggestion
---
{{- end }}
```

**Pull request merged**